modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/compaction/Compactor.java

```diff
@@ -348,6 +374,10 @@ void mergeDeltaFileToMainFile(
             return;
         }
 
+        if (filePageStore.isMarkedToDestroy()) {
```

Comment:
These checks (whether it's cancelled, whether the store is marked for destroy) are always done together in this method. How about uniting them in a single method like `shouldNotProceedWith(FilePageStore)` and call it in all places in this method?

Reply:
But for what?

Reply:
To make sure that we always do these checks together in this method. After modifications someone might accidentally add just one check instead of 2.

Reply:
I think we don't need a separate method that just calls another method, a bit redundant.